source/geometries/Next100.cc

```diff
@@ -130,49 +136,49 @@ namespace nexus {
       vessel_->GetInternalPhysicalVolume();
     G4ThreeVector vessel_displacement =
       shielding_->GetAirDisplacement(); // explained below
-    gate_zpos_in_vessel_ = vessel_->GetELzCoord();
+
+    coord_origin_ = G4ThreeVector(fc_displ_x_, fc_displ_y_, vessel_->GetGateZpos());
 
     // SHIELDING
+    shielding_->SetCoordOrigin(coord_origin_);
     shielding_->Construct();
-    shielding_->SetELzCoord(gate_zpos_in_vessel_);
     G4LogicalVolume* shielding_logic     = shielding_->GetLogicalVolume();
     G4LogicalVolume* shielding_air_logic = shielding_->GetAirLogicalVolume();
 
-    // Recall that airbox is slighly displaced in Y dimension. In order to avoid
-    // mistmatch with vertex generators, we place the vessel i
-    // n the center of the world volume
+    // Recall that air box is slightly displaced in Y dimension.
+    // In order to avoid mistmatch with vertex generators,
+    // we place the vessel in the center of the world volume
     new G4PVPlacement(0, -vessel_displacement, vessel_logic,
                       "VESSEL", shielding_air_logic, false, 0);
 
     // INNER ELEMENTS
     inner_elements_->SetLogicalVolume(vessel_internal_logic);
     inner_elements_->SetPhysicalVolume(vessel_internal_phys);
-    inner_elements_->SetELzCoord(gate_zpos_in_vessel_);
+    inner_elements_->SetCoordOrigin(coord_origin_);
     inner_elements_->SetELtoSapphireWDWdistance(gate_sapphire_wdw_distance_);
     inner_elements_->SetELtoTPdistance         (gate_tracking_plane_distance_);
     inner_elements_->Construct();
 
     // INNER COPPER SHIELDING
     ics_->SetLogicalVolume(vessel_internal_logic);
-    ics_->SetELzCoord(gate_zpos_in_vessel_);
+    ics_->SetCoordOrigin(coord_origin_);
     ics_->SetELtoSapphireWDWdistance(gate_sapphire_wdw_distance_);
     ics_->SetELtoTPdistance         (gate_tracking_plane_distance_);
     ics_->SetPortZpositions(vessel_->GetPortZpositions());
     ics_->Construct();
 
-    G4ThreeVector gate_pos(0., 0., -gate_zpos_in_vessel_);
     if (lab_walls_){
       G4ThreeVector castle_pos(0., hallA_walls_->GetLSCHallACastleY(),
                                hallA_walls_->GetLSCHallACastleZ());
 
       new G4PVPlacement(0, castle_pos, shielding_logic,
                         "LEAD_BOX", hallA_logic_, false, 0);
-      new G4PVPlacement(0, gate_pos - castle_pos, hallA_logic_,
+      new G4PVPlacement(0, -coord_origin_ - castle_pos, hallA_logic_,
                         "Hall_A", lab_logic_, false, 0, false);
     }
     else {
-      new G4PVPlacement(0, gate_pos, shielding_logic, "LEAD_BOX", lab_logic_,
-                        false, 0);
+      new G4PVPlacement(0, -coord_origin_, shielding_logic,
```

Comment:
Because if the field cage is displaced by a quantity from the origin, you want to place the whole volume that contains everything in `-` that quantity, to have the field cage in `(0, 0, 0)` as a result.

Reply:
I understand now.